endpoints/api/namespacequota.py

```diff
@@ -75,252 +74,322 @@ class OrganizationQuota(ApiResource):
+class OrganizationQuota(ApiResource):
+    schemas = {
+        "UpdateOrgQuota": {
+            "type": "object",
+            "description": "Description of a new organization quota",
+            "properties": {
+                "limit_bytes": {
+                    "type": "integer",
+                    "description": "Number of bytes the organization is allowed",
+                },
+            },
+        },
+    }
 
-        if not superperm.can():
+    @nickname("getOrganizationQuota")
+    def get(self, orgname, quota_id):
+        orgperm = OrganizationMemberPermission(orgname)
+        if not orgperm.can():
             raise Unauthorized()
 
-        quota_data = request.get_json()
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
 
-        quota = model.namespacequota.get_namespace_quota(namespace)
+        return quota_view(quota)
 
-        if quota is None:
-            msg = "quota does not exist"
-            raise request_error(message=msg)
+    @nickname("changeOrganizationQuota")
+    @validate_json_request("UpdateOrgQuota")
+    def put(self, orgname, quota_id):
+        orgperm = AdministerOrganizationPermission(orgname)
+
+        if not features.SUPER_USERS or not SuperUserPermission().can():
+            if not orgperm.can():
+                raise Unauthorized()
+
+        quota_data = request.get_json()
+
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
 
         try:
-            model.namespacequota.change_namespace_quota(namespace, quota_data["limit_bytes"])
-            return "Updated", 201
+            if "limit_bytes" in quota_data:
+                limit_bytes = quota_data["limit_bytes"]
+                model.namespacequota.update_namespace_quota_size(quota, limit_bytes)
         except model.DataModelException as ex:
             raise request_error(exception=ex)
 
-    @nickname("deleteOrganizationQuota")
-    def delete(self, namespace):
-        superperm = SuperUserPermission()
+        return quota_view(quota)
 
-        if not superperm.can():
-            raise Unauthorized()
-
-        quota = model.namespacequota.get_namespace_quota(namespace)
+    @nickname("deleteOrganizationQuota")
+    def delete(self, orgname, quota_id):
+        orgperm = AdministerOrganizationPermission(orgname)
 
-        if quota is None:
-            msg = "quota does not exist"
-            raise request_error(message=msg)
+        if not features.SUPER_USERS or not SuperUserPermission().can():
+            if not orgperm.can():
+                raise Unauthorized()
 
-        try:
-            success = model.namespacequota.delete_namespace_quota(namespace)
-            if success == 1:
-                return "Deleted", 201
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+        model.namespacequota.delete_namespace_quota(quota)
```

Comment:
If the quota is missing: 404.
Otherwise, whether the instance is deleted or not (if for some reason it's missing, we've reached our desired state, and return 204. Anything else is unexpected.

Reply:
That is certainly not obvious in reading the code, can we make it more clear?